src/main/java/org/mockito/BaseArgumentMatcher.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2017 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito;
+
+import org.mockito.internal.util.Decamelizer;
+
+/**
+ * Base class for classes that implement {@link ArgumentMatcher}.
+ *
+ * <p>This was provided to simplify migration from pre 2.x implementations; it provides a default
+ * implementation of {@link #toString()} that provides equivalent behavior to the pre 2.x version.
+ * It also resolves
+ */
+@Incubating
```

Comment:
Do we also want to make this deprecated, to signal to developers that this is an interim/temporary solution?